cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -199,17 +199,54 @@ func generateMountsModeRun() error {
 	if err != nil {
 		return err
 	}
-	// 2.2 mount base
+
+	// 2.2.1 check base status and try base if we are updating the base snap
+	baseSnapPath := ""
+	switch modeEnv.BaseStatus {
+	case "try":
+		// then we are trying a base snap update and there should be a try_base
+		// set in the modeenv too
+		if modeEnv.TryBase != "" {
+			// check that the TryBase exists in ubuntu-data
+			tryBaseSnapPath := filepath.Join(dataDir, "system-data", dirs.SnapBlobDir, modeEnv.TryBase)
+			if osutil.FileExists(tryBaseSnapPath) {
+				// set the TryBase and mount it
+				// TODO:UC20: should we write the modeenv _first_ here then
+				// output to the initramfs to create this mount for maximum
+				// reboot resiliency?
+				modeEnv.BaseStatus = "trying"
+				baseSnapPath = tryBaseSnapPath
+			} else {
+				return fmt.Errorf("try base snap %q does not exist", modeEnv.TryBase)
+			}
+		} else {
+			// TODO:UC20: should we log a message here about missing the
+			// try_base?
+			return fmt.Errorf("base status is try but no try base is set")
```

Comment:
IMO we should trigger the base snap rollback here and in the error case above. There's still a question whether we can log anything persistently, probably not.

Reply:
Yes, I left the TODOs in about logging in case there is something we can do eventually, but for now we fallback to base in every case that we can.